### PR TITLE
Add function to deeply map over nested objects and transform nested id's

### PR DESCRIPTION
**Pull request by SamGodwin2**
### What is the context of this PR?
Added a dependency on the deep-map function that allows us to easily recursively map over deeply nested objects.  

### How to review 
Changes make sense and when duplicating a page or section all nested ids should be updated to new consistent values.

### CODE REVIEW

File: eq-author-api/package.json [hunk 1]
```diff
@@ -16,7 +16,9 @@
     "body-parser": "latest",
     "cheerio": "latest",
     "cors": "latest",
+    "deep-map": "latest",
```
Comment (adammahmood) [inline eq-author-api/package.json]: I'm unsure about this library.  It's untested and hasn't been updated for 3 years.  Can we implement this algorithm ourselves?

File: eq-author-api/package.json [hunk 2]
```diff
@@ -16,7 +16,9 @@
...
     "dotenv": "latest",
+    "es6-weak-map": "^2.0.2",
```
Comment (adammahmood) [inline eq-author-api/package.json]: Why is this needed?

File: eq-author-api/package.json (hunk 1 above)
Comment (tgandrews) [reply eq-author-api/package.json]: Looks like it was updated 2 days ago to version 2? https://www.npmjs.com/package/deep-map

Comment (SamGodwin2) [reply eq-author-api/package.json]: https://www.npmjs.com/package/deep-map seems to be plenty of weekly downloads and was updated 2 days ago?

Comment (adammahmood) [reply eq-author-api/package.json]: My bad. Was looking at a different package: https://www.npmjs.com/package/deepmap

File: eq-author-api/schema/resolvers/base.js [hunk 3]
```diff
@@ -153,6 +154,24 @@ const createQuestionnaire = input => ({
   ...input,
 });
 
+const remapAllNestedIds = entity => {
```
Comment (tgandrews) [inline eq-author-api/schema/resolvers/base.js]: Can we put this in utils with a test around it?

Comment (adammahmood) [reply eq-author-api/schema/resolvers/base.js]: We need to relocate loads of methods which are located in `base.js`.  I think it's best to save this until we restructure and figure out the final folder structure of the API.

**Comment by SamGodwin2** [comment]
@adammahmood think my last comment got lost somewhere but as for the weak-map package it seems to be an issue that appeared when deep-map was updated from 1.5 to 2 so for now it might be best to pin it to 1.5 until that gets looked at.

**Comment by tgandrews** [comment]
> @adammahmood think my last comment got lost somewhere but as for the weak-map package it seems to be an issue that appeared when deep-map was updated from 1.5 to 2 so for now it might be best to pin it to 1.5 until that gets looked at.

Can we open an issue with them as it looks like it happened recently. https://github.com/mcmath/deep-map/commit/1a98c4e93e16f2dd1cbb758835271b6205568c41
